sdlib/d/gc/tcache.d

```diff
@@ -412,9 +412,16 @@ private:
 		// We trigger the de-allocation event first as it might
 		// recycle the bin we are interested in, which increase
 		// our chances that free works.
-		triggerDeallocationEvent(binInfos[sc].slotSize);
+		auto slotSize = binInfos[sc].slotSize;
+		triggerDeallocationEvent(slotSize);
 
 		auto index = getBinIndex(sc, pd.containsPointers);
+
+		// If the allocation contains pointers, zero it before freeing it
+		if (pd.containsPointers) {
+			memset(ptr, 0, slotSize);
+		}
```

Comment:
This seems like this should go directly after fetching slotSize, so we do not interleave slotSize and index related work. This ordering doesn't make much sense.

This would warrant a test and could be a patch in itself.

Reply:
Also it's not clear how large the benefit is. What about gating this with some enum we can switch on/off so we can see what the impact is?

Reply:
ok

Reply:
done